yt_dlp/extractor/franceculture.py

```diff
@@ -61,65 +31,68 @@ def _real_extract(self, url):
+                webpage,
+                'title',
+                default=self._og_search_title(webpage),
+            ),
             'description': self._html_search_regex(
-                r'(?s)<div[^>]+class="excerpt"[^>]*>(.*?)</div>', webpage, 'description', default=None),
+                r'(?s)<meta name="description" content="([^"]+)',
+                webpage,
+                'description',
+                default=None,
+            ),
             'thumbnail': self._og_search_thumbnail(webpage),
             'uploader': self._html_search_regex(
-                r'(?s)<span class="author">(.*?)</span>', webpage, 'uploader', default=None),
-            'upload_date': unified_strdate(self._html_search_regex(
-                r'(?s)class="teaser-text-date".*?(\d{2}/\d{2}/\d{4})', webpage, 'date', default=None)),
+                r'(?s)<span class="author">(.*?)</span>',
+                webpage,
+                'uploader',
+                default=None,
+            ),
+            'upload_date': ''.join(
+                re.search(
+                    r'"datePublished":"(\d{4})-(\d{2})-(\d{2})T',  # type: ignore
+                    webpage,  # type: ignore
+                ).groups()
+            ),
         }
 
-        playlist_data = self._search_regex(
-            r'''(?sx)
-                <section[^>]+data-xiti-place="[^"]*?liste_episodes[^"?]*?"[^>]*>
-                (.*?)
-                </section>
-            ''',
-            webpage, 'playlist data', fatal=False, default=None)
-
-        if playlist_data:
-            entries = []
-            for item, item_description in re.findall(
-                    r'(?s)(<button[^<]*class="[^"]*replay-button[^>]*>).*?<p[^>]*class="[^"]*teaser-text-chapo[^>]*>(.*?)</p>',
-                    playlist_data):
-
-                item_attributes = extract_attributes(item)
-                entries.append({
-                    'id': item_attributes.get('data-emission-uuid'),
-                    'url': item_attributes.get('data-url'),
-                    'title': item_attributes.get('data-diffusion-title'),
-                    'duration': int_or_none(traverse_obj(item_attributes, 'data-duration-seconds', 'data-duration-seconds')),
-                    'description': item_description,
-                    'timestamp': int_or_none(item_attributes.get('data-start-time')),
-                    'thumbnail': info['thumbnail'],
-                    'uploader': info['uploader'],
-                })
-
-            return {
-                '_type': 'playlist',
-                'entries': entries,
-                **info
-            }
+        video_data = json.loads(
+            self._search_regex(  # type: ignore
+                r'({"@type":"AudioObject","contentUrl":"[^"]+","duration":"[^"]+","encodingFormat":"mp3","potentialAction":{"@type":"Action","name":"ListenAction"}})',
+                webpage,
+                'video data',
+            )
+        )
```

Comment:
```suggestion
        video_data = self._parse_json(
            self._search_regex(
                 r'({"@type":"AudioObject","contentUrl":"[^"]+","duration":"[^"]+","encodingFormat":"mp3","potentialAction":{"@type":"Action","name":"ListenAction"}})', webpage, 'video data'),
            video_id)
```

Don't use `json` module in extractor code.

Reply:
And why not just capture whole `<script type="application/ld+json">` tags and comb them for metadata?

Reply:
We have a util function for just that `self._search_json_ld`

Reply:
so it'll be `traverse_obj(self._search_json_ld(...), (lambda _,v: v['@typ?e']=='AudioObject'))`

Reply:
There are several `ld+json` tags in the webpage. `self._search_json_ld` returns just one of them, and it's not the one that is relevant.

Reply:
From what I can tell skimming the code, `_search_json_ld` attempts to catch all JSON-LD tags and aggregate them into a single dict. I cannot say how sensible this approach is.

Reply:
Use `expected_type='AudioObject'`

Reply:
**Two things here**:

1/ It looks like this [`break` statement](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L1505=) has the unfortunate consequence that when several `json_ld`s are found, if one of them is not top level and `traverse_json_ld` is called recursively, subsequent `json_ld`s are skipped. Is that on purpose?

- If yes, this implementation of `traverse_json_ld` doesn't seem to be compatible with what's required for this extractor: the first `json_ld` is not top level, and the relevant information is in the second one the parsing of which never happens because of this `break`.
- If not, the `break` can be replaced by a `continue` to parse the remaining `json_ld`s, but IMHO it's beyond the scope of this PR, and I hit a second problem anyway:

2/ Looks like `common.py`'s `traverse_json_ld` has built-in support for [`TVEpisode`, `Episode`](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L1512=), [`Movie`](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L1530=), [`Article`, `NewsArticle`](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L1537=) and [`VideoObject`](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L1547=). If it's not one of them, then [a last attempt is made to extract the video](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L1553-L1555=).

However, this particular `json_ld` that's relevant for this extractor is a `RadioEpisode`, the content of which is the following:

```json
{
  "@type": "RadioEpisode",
  "name": "La physique d’Einstein aiderait-elle à comprendre le cerveau ?",
  "description": "Existerait-il un pont conceptuel entre la physique de l’espace-temps et les neurosciences ?",
  "dateCreated": "2022-05-14T14:00:00.000Z",
  "inLanguage": "fr",
  "isAccessibleForFree": "True",
  "mainEntity": {
    "@type": "AudioObject",
    "contentUrl": "https://media.radiofrance-podcast.net/podcast09/23118-14.05.2022-ITEMA_23026189-2022C22785S0134-21.mp3",
    "duration": "P0Y0M0DT0H45M50S",
    "encodingFormat": "mp3",
    "potentialAction": {
      "@type": "Action",
      "name": "ListenAction"
    }
  },
  "image": {
    "@type": "ImageObject",
    "url": "https://cdn.radiofrance.fr/s3/cruiser-production/2022/05/d184e7a3-4827-4494-bf94-04ed7b120db4/2048x640_gettyimages-200171095-001.jpg",
    "width": "1891",
    "height": "590",
    "copyrightHolder": "Getty"
  },
  "director": [
    {
      "@type": "Person",
      "name": "Etienne Klein"
    }
  ],
  "partOfSeries": {
    "@type": "RadioSeries",
    "name": "La physique d’Einstein aiderait-elle à comprendre le cerveau ?",
    "url": "https://www.radiofrance.fr/franceculture/podcasts/science-en-questions/la-physique-d-einstein-aiderait-elle-a-comprendre-le-cerveau-8440487"
  }
}
```

Not being one of the supported type, and not having a `video` key, leads the parsing reach the next [break statement](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L1559=), and the content is not extracted as expected.

A solution could be to implement a handler for `RadioEpisode` in `traverse_json_ld`. However, I don't think it makes sense to have extractor specific code (if it is?) in `common.py`, not to mention that's most likely way too deep in the internals of `yt-dlp` for my current knowledge. Moreover, it's also beyond the scope of this PR.

**Questions**:

- Is my understanding above correct?
- Should two other PRs come before that extractor can leverage `_search_json_ld`, one for the `break` → `continue` (at the risk of breaking other existing extractors), and another one to add a `RadioEpisode` handler that seems to be specific of this FranceCulture extractor so far, or
- Should the current PR be merged as is, or modified?

Reply:
Thanks for the detailed explanation.

`RadioEpisode` is a part of the schema. So if that were the only issue, we could add it. But idk whether we can fix (1) without breaking other extractors. So it seems best to not use `traverse_json_ld` for the time-being

Reply:
Let's stick with the current PR implementation with `self._parse_json`, then? If so, I hope this PR can be merged?